test/model/avatar_url_test.dart

```diff
@@ -0,0 +1,147 @@
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+import 'package:zulip/model/avatar_url.dart';
+
+class ExampleUser {
+  final Uri realm;
+  final int userId;
+  final String email;
+
+  ExampleUser({
+    required this.realm,
+    required this.userId,
+    required this.email,
+  });
+}
+
+void main() {
+  var egUser = ExampleUser(
+    realm: Uri.parse("https://zulip.example.org"),
+    userId: 1417,
+    email: "sample@zulip.org",
+  );
+
+  double defaultSz = 30;
+  double largeSz = 120;
+
+  group('GravatarURL', () {
+    test('smoke with null url', () {
+      check(AvatarURL.fromUserOrBotData(
+        realm: egUser.realm,
+        userId: egUser.userId,
+        email: egUser.email,
+      ))
+      .has((obj) => obj.get(defaultSz), 'computed uri')
+      .has((uri) =>
+        uri.toString().contains(GravatarURL.origin),
+        'verify gravatar',
+      )
+      .equals(true);
+    });
+
+    test('smoke with empty string url', () {
+      check(AvatarURL.fromUserOrBotData(
+        rawAvatarUrl: '',
+        realm: egUser.realm,
+        userId: egUser.userId,
+        email: egUser.email,
+      ))
+      .has((obj) => obj.get(defaultSz), 'computed uri')
+      .has((uri) =>
+        uri.toString().contains(GravatarURL.origin),
+        'verify gravatar',
+      )
+      .equals(true);
+    });
+
+    test('smoke with gravatar url', () {
+      String avatarUrl = '${GravatarURL.origin}/avatar/1234';
+      check(AvatarURL.fromUserOrBotData(
+        rawAvatarUrl: avatarUrl,
+        realm: egUser.realm,
+        userId: egUser.userId,
+        email: egUser.email,
+      ))
+      .has((obj) => obj.get(defaultSz), 'computed uri')
+      .has((uri) =>
+        uri.toString() == '$avatarUrl?s=30.0',
+        'verify gravatar',
+      )
+      .equals(true);
+    });
+  });
+
+  group('UploadedAvatarURL', () {
+    test('smoke with png image url', () {
+      String avatarUrl = 'https://zulip.example/image.png';
+      check(AvatarURL.fromUserOrBotData(
+        rawAvatarUrl: avatarUrl,
+        realm: egUser.realm,
+        userId: egUser.userId,
+        email: egUser.email,
+      ))
+      .has((obj) => obj.get(defaultSz), 'computed uri')
+      .has((uri) =>
+        uri.toString() == avatarUrl,
+        'verify avatar',
+      )
+      .equals(true);
+    });
```

Comment:
Let's use `check` in the following style instead:
```suggestion
  group('UploadedAvatarURL', () {
    test('png image url', () {
      const url = 'https://zulip.example/image.png';
      final avatarUrl = AvatarURL.fromUserOrBotData(
        rawAvatarUrl: url,
        realm: egUser.realm,
        userId: egUser.userId,
        email: egUser.email,
      );
      check(avatarUrl.get(defaultSz).toString()).equals(url);
    });
```

In general `has` is good for defining extension getters on `Subject`, as seen in many examples in our test suite, but rarely inside individual test cases.

Also as that illustrates:
* this is straightforwardly a unit test of this code, not a [smoke test](https://en.wikipedia.org/wiki/Smoke_testing_(software)), so doesn't need the qualifier "smoke" in the name
* local variables are final or const if there's no intention to mutate them
* local variables don't need type annotations unless there's a reason type inference wouldn't give the same answer

Reply:
[ecec588](https://github.com/zulip/zulip-flutter/pull/534/commits/ecec58836637ac686e88dade6d241dbd31bcbefc)